project-addons/custom_account/models/purchase.py

```diff
@@ -35,3 +36,12 @@ def button_cancel(self):
     end_manufacture = fields.Date("Ending Date Of Manufacture")
     sale_notes = fields.Text("Purchase Sale Notes")
     remark = fields.Char("Remark")
+    send_date_planned_to_lines = fields.Boolean("Set date to all order lines",default=True)
+
+    def button_confirm(self):
+        for line in self.order_line:
+            if line.product_id.default_code =="----- PTE NOMBRE -----":
+                raise UserError(_("An order cannot be confirmed with a product called \"")+line.product_id.default_code+"\"")
+        if self.send_date_planned_to_lines:
```

Comment:
este método es una api.multi mete toda la lógica dentro de un for purchase in self:

Reply:
Ya está corregido